### PR TITLE
chore(TCOMP-831): expose getSupportedProducts to Datastore

### PR DESCRIPTION
**Pull request by undx**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TCOMP-831

**What is the new behavior?**

- change Datastore interface
- include supportedProducts to DTO

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [X] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by undx** [comment]
Actually, I'm using java 8 in `components-common` for providing a default interface behavior and not breaking the whole build. 
But, after validation, we'll remove this and force implementaton of the `getSupportedProducts` in datastores.

File: services/components-api-service-rest/src/main/java/org/talend/components/service/rest/dto/DefinitionDTO.java [hunk 1]
```diff
@@ -100,6 +103,7 @@ public DefinitionDTO(DatastoreDefinition origin) {
         this.type = "datastore";
         this.inputCompName = origin.getInputCompDefinitionName();
         this.outputCompName = origin.getOutputCompDefinitionName();
+        this.supportedProducts = origin.getSupportedProducts();
```
Comment (gonchar-ivan) [inline services/components-api-service-rest/src/main/java/org/talend/components/service/rest/dto/DefinitionDTO.java]: there is common Constructor for both Datastore/Component definition.
```
private DefinitionDTO(Definition origin)
```
However, I found it is not easy to move the code there:
`this.supportedProducts = origin.getSupportedProducts();`
as Definition interface has no getSupportedProducts(); method
So, I agree with current implementation

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/24/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/24/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTCOMP-831_ExposeGetSupportedProductsToDatastore)

**Comment by undx** [comment]
feature cancelled

**Pull request closed without merging**